Use common state for cloned view + Immutable set of paceholder salt + Improve fluent interface + Fix configs (#195)
yiisoft/view: README.md

- [Basic functionality](docs/basic-functionality.md) for use in any environment.
- Advanced functionality for [use in a WEB environment](docs/use-in-web-environment.md).

### State of `View` and `WebView` services

> State of `View` service:
> - parameters,
> - blocks.
> 
> State of `WebView` service:
> - parameters,
> - blocks,
> - title,
> - meta and link tags,
> - JS/CSS strings,
> - JS/CSS files.

Reviewer: ```suggestion
While being immutable and, by itself, stateless, both `View` and `WebView` services have sets of stateful and mutable data.

`View` service:
- parameters,
- blocks.

`WebView` service:
- parameters,
- blocks,
- title,
- meta and link tags,
- JS/CSS strings,
- JS/CSS files.
```

> State of `View` and `WebView` services don't clone on clone of service instances, including when calling methods
> `with*()`. This allows use one state for cloned services. For example, we can get `WebView` from container in 
> controller via dependency injection and change context path:    

Reviewer: ```suggestion
The state of `View` and `WebView` is not cloned when the services are cloned. So when
using `with*()`, both new and old instances are sharing the same set of stateful mutable data. It allows, for example, to get `WebView` via type-hinting in a controller and change context path:
```

```php
final class BlogController {
    private WebView $view;
    public function __construct (WebView $view) {
        $this->view = $view->withContextPath(__DIR__.'/views');
    }
}
```

> ... and then register CSS in widget:

Reviewer: ```suggestion
... and then register CSS in a widget:
```

```php
final class LastPosts extends Widget 
{    
    private WebView $view;
    public function __construct (WebView $view) {
        $this->view = $view;
    }
    protected function run(): string
    {
        ...
        $this->view->registerCss('.lastPosts { background: #f1f1f1; }');
        ...
    }
}
```

> For full clone of `View` or `WebView` service use method `withClearedState()` that also clear and clone state: 

Reviewer: ```suggestion
To get a deep cloned `View` or `WebView` use `withClearedState()`: 
```
